src/sidebar/store/modules/annotations.ts

```diff
@@ -3,46 +3,85 @@
  * sidebar.
  */
 
+import type { Dispatch } from 'redux';
 import { createSelector } from 'reselect';
 
 import { hasOwn } from '../../../shared/has-own';
+import type { Annotation, SavedAnnotation } from '../../../types/api';
+import type { HighlightCluster } from '../../../types/shared';
+
 import * as metadata from '../../helpers/annotation-metadata';
 import { isHighlight, isSaved } from '../../helpers/annotation-metadata';
 import { countIf, toTrueMap, trueKeys } from '../../util/collections';
 import { createStoreModule, makeAction } from '../create-store';
 
 import { routeModule } from './route';
+import type { State as RouteState } from './route';
 import { sessionModule } from './session';
+import type { State as SessionState } from './session';
 
-/**
- * @typedef {'anchored'|'orphan'|'timeout'} AnchorStatus
- * @typedef {import('../../../types/api').Annotation} Annotation
- * @typedef {import('../../../types/shared').HighlightCluster} HighlightCluster
- * @typedef {import('../../../types/shared').ClientAnnotationData} ClientAnnotationData
- * @typedef {import('../../../types/api').SavedAnnotation} SavedAnnotation
- */
+type AnchorStatus = 'anchored' | 'orphan' | 'timeout';
 
-/**
- * @typedef AnnotationStub
- * @prop {string} [id] - service-provided identifier if annotation has been
- *       persisted to the service
- * @prop {string} [$tag] - local-generated identifier
- */
+type AnchorStatusUpdates = {
+  [$tag: string]: AnchorStatus;
```

Comment:
Here and the `initialState` props `highlighted` and `hovered` reflect my own frustration when I can't see quickly whether a type operates on a `$tag` or `id` in these maps because it's typed as `Record<string, boolean>`. This syntax ensures that _wherever_ and _however_ my editor gives me a type hint, it's going to let me know we're talking about `$tags` here.